### PR TITLE
fix(CKEditor): Fix config.toolbar type

### PR DESCRIPTION
**Pull request by adrienverge**
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] adds corresponding tests to the test file.

---

The type for toolbar is currently defined as  `string | (string[])[]`,
but it should be `string | (string | string[])[]`.

The `toolbar` property is either:
- a toolbar name (`string`)
- or an array of elements, that can be `string`s (for instance a newline
  marker `'/'`) or `string[]`s (for instance a list of buttons).

For example, this is a valid configuration:

``` js
CKEDITOR.config.toolbar = [
    [ 'mode', 'document', 'doctools' ],
    [ 'clipboard', 'undo' ],
    '/',
    [ 'find', 'selection', 'spellchecker' ],
    [ 'basicstyles', 'cleanup' ],
    '/',
    [ 'list', 'indent', 'blocks', 'align', 'bidi' ],
];
```

### CODE REVIEW

**Comment by vladima** [comment]
per [CKEditor.config.toolbar documentation](http://docs.ckeditor.com/#!/api/CKEDITOR.config-cfg-toolbar)
>toolbar : Array/String
The toolbox (alias toolbar) definition. It is a toolbar name or an array of toolbars (strips), each one being also an array, containing a list of UI items.

which is exactly what original typings used to be. Can you please clarify if this is a typo in CKEditor documentation?

**Comment by adrienverge** [comment]
Thanks for reviewing @vladima.

Yes, this is probably a typo in documentation (I saw many others). I confirm the type for toolbar should be `string | (string | string[])[]`, which means `'/'` can be inserted among rows to put a break between them. I've tested it (I actually use it); and it can be seen directly in the source code:
https://github.com/ckeditor/ckeditor-dev/blob/12f0de3/plugins/toolbar/plugin.js#L216-L221

In other words, `toolbar` can be either:
```js
CKEDITOR.config.toolbar = 'Basic';
```
or:
```js
CKEDITOR.config.toolbar = [
    [ 'mode', 'document', 'doctools' ],
    '/',
    [ 'find', 'selection', 'spellchecker' ],
    [ 'basicstyles', 'cleanup' ],
];
```

**Comment by adrienverge** [comment]
For information, I submitted a pull request to clarify documentation: https://github.com/ckeditor/ckeditor-dev/pull/296

Although they don't want to merge "to keep the documentation simple", **they confirm this** `'/'` **option exists and is valid**.
